### PR TITLE
Remove redundant epoch_number variable in training tutorial (issue 3714)

### LINKED ISSUE #3714: Feedback about Training with PyTorch
Hi! There is the following issue on this page: https://docs.pytorch.org/tutorials/beginner/introyt/trainingyt.html

One dumb issue here! In the block code inside the Per-Epoch Activity section the `epoch_number` is just unnecessary having `epoch`. 

Thanks!

### PR DESCRIPTION
**Pull request by sheralskumar**
Fixes #3714 

## Description
Removed redundant epoch_number variable and used the loop variable epoch directly for logging and checkpointing. This simplifies the code, improving clarity for beginners.

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

**Comment by sekyondaMeta** [comment]
@claude

**Comment by sheralskumar** [comment]
It looks like one built test failed due to an unrelated error in intermediate_source/parametrizations.py (singular matrix in torch.linalg.solve).

Could a maintainer please re-run the CI? Thank you!
